src/rgw/rgw_sal.h

```diff
@@ -1096,6 +1096,9 @@ class Object {
     /** Get a unique copy of this object */
     virtual std::unique_ptr<Object> clone() = 0;
 
+    virtual jspan_context& get_trace() = 0;
+    virtual void set_trace (jspan_context&& _trace_ctx) = 0;
```

Comment:
is `SpanContext` really what we want to pass around in rgw? the context is just the piece that gets serialized over the network, so makes sense for the librados API. but if we want to add events or child spans to this trace, we'd need a pointer or shared_ptr to the `Span` itself

Reply:
you are right. will do in a followup PR